### ISSUE TITLE
Easily adding one-hot encoding

### ISSUE THREAD

**Issue by dorienh** [jdb78/pytorch-forecasting#1457]
Thanks for the great library! 

I was wondering if there is a straightforward way to apply one-hot encoding to the categorical varibales: 

```
self.training_dataset = TimeSeriesDataSet(
  ...
  time_varying_known_categoricals=['pair'],
  ...
  categorical_encoders={'pair': pytorch_forecasting.data.encoders.NaNLabelEncoder(add_nan=True).fit(dataset.pair)}, ]
```

Instead of just NanLabelEncoder is it possible to somehow apply one-hot encoding here?

**Comment by manitadayon** [comment]
@dorienh can you elaborate more? If a features takes 170 different values you do not want one-hot encoding anyway, you want to do embedding. If you have 170 different features you can add them in a loop. Any example will clear the problem more.

**Comment by manitadayon** [comment]
No, you just include them as the static categorical variables or time varying categorical variables, then the packages will perform the embedding automatically. No need for any embedding. Add static categorical or time varying categorical in the TimeSeriesDataSet.

**Comment by dorienh** [comment]
I am building my own fully custom model though, only using TimeSeriesDataset(). When I look at x["encoder_cat"], the it's only of length 1, i.e. it seems to map strings to range(1,length).
